### PR TITLE
Random order symmetric peers

### PR DESCRIPTION
**Pull request by marc-casperlabs**
Ensure we only select symmetrically connected peers when syncing.

### CODE REVIEW

File: node/src/components/small_network.rs [hunk 1]
```diff
@@ -860,10 +860,16 @@ where
...
-                    peers_vec.shuffle(rng);
-                    responder.respond(peers_vec).ignore()
+                NetworkInfoRequest::GetFullyConnectedPeers { responder } => {
+                    let symmetric_peers: Vec<NodeId> = self
+                        .connection_symmetries
+                        .iter()
+                        .filter_map(|(node_id, sym)| {
+                            matches!(sym, ConnectionSymmetry::Symmetric { .. }).then(|| *node_id)
+                        })
+                        .collect();
```
Comment (goral09) [inline node/src/components/small_network.rs:870]: Why not add a method on `enum ConnectionSymmetries` to return a collection of symmetric connections?

File: node/src/effect.rs [hunk 2]
```diff
@@ -624,15 +624,15 @@ impl<REv> EffectBuilder<REv> {
...
+    pub async fn get_fully_connected_peers<I>(self) -> Vec<I>
     where
         REv: From<NetworkInfoRequest<I>>,
         I: Send + 'static,
     {
         self.make_request(
-            |responder| NetworkInfoRequest::GetPeersInRandomOrder { responder },
-            QueueKind::Api,
+            |responder| NetworkInfoRequest::GetFullyConnectedPeers { responder },
+            QueueKind::Regular,
```
Comment (afck) [inline node/src/effect.rs:635]: Good catch! `Api` made the priority much lower, did it?

**Review by goral09 — APPROVED** [review]

**Review by afck — APPROVED** [review]

File: node/src/effect.rs (hunk 2 above)
Comment (marc-casperlabs) [reply node/src/effect.rs:635]: In a way. It made it share resources with any external API requests.

File: node/src/components/small_network.rs (hunk 1 above)
Comment (marc-casperlabs) [reply node/src/components/small_network.rs:870]: There's no `ConnectionSymmetries`. The `ConnectionSymmetry` enum holds only connections for a single peer. The only method here would be something like `ConnectionSymmetries::is_symmetric`, which would contain the `matches!`. I opted against that for now, I'd add if I needed the functionality again.

Comment (goral09) [reply node/src/components/small_network.rs:870]: Ah, b/c we want to extract the `node_id`s that are keys in the hashmap of `connection_symmetries` in which a value is of type `ConnectionSymmetry` 😵

**Review by afck — APPROVED** [review]

**Review by Fraser999 — APPROVED** [review]

**Review by afck — APPROVED** [review]
Also looks fine to me; but why did you switch to this approach?

**Review by Fraser999 — APPROVED** [review]

**Comment by goral09** [comment]
bors retry

**Comment by goral09** [comment]
bors retry

**Pull request merged**